File: src/EventFlow.Domain/Entities/OrderAggregate/Order.cs
namespace EventFlow.Domain.Entities.OrderAggregate;

public class Order : BaseEntity
{
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public string BuyerEmail { get; set; } = default!;
    public PaymentSummary PaymentSummary { get; set; } = null!;
    public List<OrderItem> OrderItems { get; set; } = [];
    public decimal Subtotal { get; set; }
    public decimal Discount { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public required string PaymentIntentId { get; set; }

    public decimal GetTotal()
    {
        return Subtotal - Discount;
    }
}

File: client/event-flow/src/app/shared/models/order.ts
export type Order = {
    id: string
    orderDate: string
    buyerEmail: string
    paymentSummary: PaymentSummary
    orderItems: OrderItem[]
    subtotal: number
    discount?: number
    status: string
    total: number
    paymentIntentId: string
}

export type PaymentSummary = {
    last4: number
    brand: string
    expMonth: number
    expYear: number
}

export type OrderItem = {
    ticketId: string
    eventName: string
    price: number
    quantity: number
}

export type OrderToCreate = {
    cartId: string;
    paymentSummary: PaymentSummary;
    discount?: number;
}